openpype/hosts/houdini/plugins/publish/validate_primitive_hierarchy_paths.py

```diff
@@ -104,3 +112,47 @@ def get_invalid(cls, instance):
                 "(%s of %s prims)" % (path_attr, len(invalid_prims), num_prims)
             )
             return [output_node.path()]
+
+    @classmethod
+    def repair(cls, instance):
+        """Repair action if `Path Attribute` wasn't exist.
+
+        It Adds a "name" node before the the output node to automatically
+        force a path attribute on the geometry. Be aware that this will
+        create a `default single name` and thus will merge all geometry in one shape.
+        """
+
+        output_node = instance.data.get("output_node")
+        rop_node = hou.node(instance.data["instance_node"])
+
+        if not (isinstance(output_node, hou.SopNode) and \
+                 output_node.type().name() == "output"):
+            # In the case someone has explicitly set an Object
+            # node instead of a SOP Output node in Geometry context
+            # then for now we ignore - this allows us to also
+            # export object transforms.
+            cls.log.warning("No geometry output node found, skipping Repair Action..")
+            return
+
+        frame = instance.data.get("frameStart", 0)
+        geo = output_node.geometryAtFrame(frame)
+
+        path_attr = rop_node.parm("path_attrib").eval() or "path"
+        paths = geo.findPrimAttrib(path_attr) and \
+                    geo.primStringAttribValues(path_attr)
+
+        if not paths:
```

Comment:
Instead of basically reimplementing the validation check again I tend to write a `get_invalid` method that both `process` and `repair` use to detect whether it's currently still invalid.

Then repair could do:
```
if not cls.get_invalid(instance):
    # all ok
    return

# would probably move this into a method too so both `get_invalid` and this use the same logic 
path_attr = cls.get_path_attr(instance)  

output_node = instance.data.get("output_node")
path_node = output_node.parent().createNode("name", "AUTO_PATH")
path_node.parm("attribname").set(path_attr)
path_node.parm("name1").set('`opname("..")`/`opoutput(".", 0)`')

path_node.setFirstInput(output_node.input(0))
path_node.moveToGoodPosition()
output_node.setFirstInput(path_node)
output_node.moveToGoodPosition()

cls.log.debug(
    "'%s' was created. It adds '%s' with a default single value"
     % (path_node, path_attr)
)
```